test/e2e/common/node/framework/cgroups/cgroups.go

```diff
@@ -443,3 +445,26 @@ func IsPodOnCgroupv2Node(f *framework.Framework, pod *v1.Pod) (result bool) {
 	// So, we need to see if "/sys/fs/cgroup" is contained in the output.
 	return strings.Contains(out, "/sys/fs/cgroup")
 }
+
+func getExpectedCPUShares(cpuRequest, cpuLimit *resource.Quantity, podOnCgroupv2 bool) []string {
+	// This function is moved out from cgroups.go because opencontainers/cgroups can only be compiled in linux platforms.
+	var shares int64
+	if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+		shares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
+	} else {
+		shares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
+	}
+	if podOnCgroupv2 {
+		// Because of https://github.com/kubernetes/kubernetes/issues/131216, the way of conversion has been changed.
+		// runc: https://github.com/opencontainers/runc/pull/4785
+		// crun: https://github.com/containers/crun/issues/1721
+		// This is dependent on the container runtime version. In order not to break the tests when we upgrade the
+		// container runtimes, we check if either the old or the new conversion matches the actual value for now.
+		// TODO: Remove the old conversion once container runtimes are updated.
+		oldConverted := 1 + ((shares-2)*9999)/262142
+		converted := libcontainercgroups.ConvertCPUSharesToCgroupV2Value(uint64(shares))
```

Comment:
Ugh, this utility is portable, but the package it is contained in is not.

We'll need to split out this code.
/hold